From caea323780508ead9778fef671eb154d38621311 Mon Sep 17 00:00:00 2001
From: Kareem Shaik <kareem.m.shaik@intel.com>
Date: Mon, 18 Sep 2017 19:13:27 +0530
Subject: [PATCH 0325/1240] ASoC: Intel: Skylake: Reconfigure host DMA in case
 of XRUN

In case of XRUN recovery, PCM prepare will be called. In this case
Host DMA is reset and reconfigured. If restart does not configure
DMA after DSP pipe is reset, it eventually results in I/O error.

Change-Id: I6de42991af26ea478da314bd9c561118aadb50d3
Signed-off-by: Kareem Shaik <kareem.m.shaik@intel.com>
Reviewed-by: Gogineni, GiribabuX <giribabux.gogineni@intel.com>
Reviewed-by: Chippalkatti, MallikarjunX <mallikarjunx.chippalkatti@intel.com>
Reviewed-by: Kale, Sanyog R <sanyog.r.kale@intel.com>
Reviewed-by: Sinha, Mohit <mohit.sinha@intel.com>
Reviewed-by: Koul, Vinod <vinod.koul@intel.com>
Tested-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
---
 sound/soc/intel/skylake/skl-pcm.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/skylake/skl-pcm.c b/sound/soc/intel/skylake/skl-pcm.c
index 86d69c9..396f417 100644
--- a/sound/soc/intel/skylake/skl-pcm.c
+++ b/sound/soc/intel/skylake/skl-pcm.c
@@ -324,8 +324,10 @@ static int skl_pcm_prepare(struct snd_pcm_substream *substream,
 
 	/* In case of XRUN recovery, reset the FW pipe to clean state */
 	if (mconfig && (substream->runtime->status->state ==
-					SNDRV_PCM_STATE_XRUN))
+					SNDRV_PCM_STATE_XRUN)) {
 		skl_reset_pipe(skl->skl_sst, mconfig->pipe);
+		skl_pcm_host_dma_prepare(dai->dev, mconfig->pipe->p_params);
+	}
 
 	return 0;
 }
-- 
2.7.4

